vars/recordDeployment.groovy

```diff
@@ -0,0 +1,51 @@
+def call(owner, repo, ref, status, environmentURL, Map userConfig=[:]) {
+  def defaultConfig = [
+    environment: "preview",
+    description: "Deploy to preview environment",
+    credentialId: "github-app-infra"
+  ]
+
+  // Merging the 2 maps - https://blog.mrhaki.com/2010/04/groovy-goodness-adding-maps-to-map_21.html
+  final Map finalConfig = defaultConfig << userConfig
+
+  withCredentials([usernamePassword(credentialsId: finalConfig.credentialsId, usernameVariable: 'GITHUB_APP', passwordVariable: 'GH_TOKEN')]) {
+    withEnv([
+        "STATUS=${status}",
+        "REF=${ref}",
+        "ENVIRONMENT=${finalConfig.environment}",
+        "DESCRIPTION=${finalConfig.description}",
+        "TRANSIENT_ENVIRONMENT=${finalConfig.environment != "production" ? "true" : "false"}",
+        "LOG_URL=${BUILD_URL}console",
+        "ENVIRONMENT_URL=${environmentURL}",
+        "OWNER=${owner}",
+        "REPO=${repo}",
+    ]) {
+      def id = sh(script: """
```

Comment:
```suggestion
      def id = sh(script: '''
```

=> WDYT about using simple quotes to avoid groovy interpolation (since you added the `withEnv` instruction) ?